### PR TITLE
Better error message when viewing a password protected map via http

### LINKED ISSUE #449: Refactor token without https error
When [this check is triggered](https://github.com/CartoDB/cartodb.js/blob/101959f7e5d7a89391a61995c27dc407cca47bff/src/geo/layer_definition.js#L759) it logs an error but page keeps loading, so new errors can be generated (see  CartoDB/cartodb#3307) which are no obviously related (well, at least, not for me :_) ). Maybe it could be more explicit, display a user error or stop further processing, not sure.

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #449. This is something that will only occur when serving cartodb (the maps API) via http. I'm assuming https is a requirement for on-premise installations, so we will only see this problem when running the app locally.

I reworded the error a bit. Instead of:

```
MAP: error creating layernamedmap Error: https must be used when auth_token is set
```

cartodb.js will spit the following error when trying to view a protected map via http:

```
MAP: error creating 'namedmap' layer -> https must be used when map has token authentication
```

Also, we're now checking if the layer view was created before trying to add legends to it, to prevent another JS error from being displayed.

cc/ @juanignaciosl

### CODE REVIEW

File: src/geo/layer_definition.js [hunk 1]
```diff
@@ -857,7 +857,7 @@ NamedMap.prototype = _.extend({}, MapBase.prototype, {
 
   setAuthToken: function(token) {
     if(!this.isHttps()) {
-      throw new Error("https must be used when auth_token is set");
+      throw new Error("https must be used when map has token authentication");
```
Comment (alonsogarciapablo) [inline src/geo/layer_definition.js]: We could even be more explicit and say something like:

```
Maps API should be served via https when making authenticated requests.
```

**Comment by alonsogarciapablo** [comment]
bump!

File: src/vis/vis.js [hunk 2]
```diff
@@ -831,10 +831,13 @@ var Vis = cdb.core.View.extend({
     var self = this;
     for (var i = layers.length - 1; i >= 0; --i) {
       var cid = layers.at(i).cid;
-      var layer = layers.at(i).attributes
+      var layer = layers.at(i).attributes;
       if (layer.visible) {
         var layerView = this.mapView.getLayerByCid(cid);
-        legends.push(this._createLayerLegendView(layer, layerView));
+        if (layerView) {
```
Comment (alonsogarciapablo) [inline src/vis/vis.js]: We were assuming the layer had been added to the mapView when adding the legends, and this was triggering a JS error.

**Comment by alonsogarciapablo** [comment]
@xavijam What do you think about this? Thanks!

**Comment by alonsogarciapablo** [comment]
This PR has been sitting here for some days now. @xavijam @juanignaciosl unless you have something to say, I will go ahead and merge it. Thanks!

**Comment by juanignaciosl** [comment]
Oh, sorry. Since changes were in front I thought you just pinged me. Nothing to say :+1:

**Comment by xavijam** [comment]
Looks good buddy, go ahead :es:

**Comment by alonsogarciapablo** [comment]
Thanks guys! :smile:

**Pull request merged**